### PR TITLE
integral changes

### PR DESCRIPTION
**Pull request by smichr**

### CODE REVIEW

**Comment by gxyd** [comment]
The code changes look good to me.

File: sympy/integrals/integrals.py [hunk 1]
```diff
@@ -435,67 +443,79 @@ def doit(self, **hints):
...
-                    (xab[1].has(oo, -oo) or xab[2].has(oo, -oo)):
-                ret = try_meijerg(function, xab)
-                if ret is not None:
-                    function = ret
-                    continue
+            if isinstance(function, Piecewise):
+                # TODO pass along eval_kwargs
+                if len(xab) == 1:
+                    antideriv = function._eval_integral(xab[0]
+                        )
```
Comment (gxyd) [inline sympy/integrals/integrals.py]: Is it necessary to put this closing bracket on next line?

**Comment by gxyd** [comment]
As you are at this. Can you see what can we make use of from PR https://github.com/sympy/sympy/pull/11797 ?

File: sympy/integrals/integrals.py (hunk 1 above)
Comment (smichr) [reply sympy/integrals/integrals.py]: That's where the `eval_kwargs` are going to go.

**Comment by smichr** [comment]
> Can you see what can we make use of from PR #11797 ?

I suspect others would be better at understanding what should be attempted when. When I was looking at this with the Piecewise lenses, I could see that it should have handled that first *and* should have passed along the keywords so when the parts are being integrated they could use the original hints. I remember thinking the whole thing looked pretty contorted. I'll see if there is anything obvious that I can address.

**Pull request merged**